test/unit/test_events.py

```diff
@@ -70,4 +74,24 @@ def test_event_codes(self):
                 # cannot have been used already
                 self.assertFalse(event.code in all_codes, f'{event.code} is assigned more than once. Check types.py for duplicates.')
                 all_codes.add(event.code)
-                
\ No newline at end of file
+
+class TestEventBuffer(TestCase):
+
+    # ensure events are populated to the buffer exactly once
+    def test_buffer_populates(self):
+        fire_event(UnitTestInfo(msg="Test Event 1"))
+        fire_event(UnitTestInfo(msg="Test Event 2"))
+        self.assertTrue(
+            EVENT_HISTORY.count(UnitTestInfo(msg='Test Event 1', code='T006')) == 1
+        )
+
+    # ensure events drop from the front of the buffer when buffer maxsize is reached
+    def test_buffer_FIFOs(self):
+        for n in range(0,100001):
```

Comment:
FYI -- This is adding ~10 seconds to the unit test run time... can be fixed by correcting TODO on `core/dbt/events/functions.py` line 24.

Reply:
ahh. making it configurable so we can test with a smaller queue.